src/core/include/openvino/op/grid_sample.hpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/attribute_adapter.hpp"
+#include "openvino/op/op.hpp"
+#include "openvino/op/util/attr_types.hpp"
+
+namespace ov {
+namespace op {
+namespace v9 {
+
+/// \brief Operator performing interpolated sampling of the input tensor
+/// \ingroup ov_ops_cpp_api
+class OPENVINO_API GridSample : public Op {
+public:
+    OPENVINO_OP("GridSample", "opset9");
+    BWDCMP_RTTI_DECLARATION;
+
+    enum class InterpolationMode { BILINEAR, BICUBIC, NEAREST };
+    enum class PaddingMode { ZEROS, BORDER, REFLECTION };
```

Comment:
How these two enums translate in Python? I do not see any tests for them and I am unsure of the results...

Reply:
Those enums are taken as strings from Python (and during the IR deserialization and translated using the `EnumNames` template specialization).